src/types/mappingproxy.rs

```diff
@@ -0,0 +1,597 @@
+    }
+
+    /// Returns a list of mappingproxy keys.
+    ///
+    /// This is equivalent to the Python expression `list(mappingproxy.keys())`.
+    pub fn keys(&self) -> &PyList {
+        unsafe {
+            PySequence::try_from_unchecked(self.call_method0("keys").unwrap())
+                .to_list()
+                .unwrap()
+        }
+    }
+
+    /// Returns a list of mappingproxy values.
+    ///
+    /// This is equivalent to the Python expression `list(mappingproxy.values())`.
+    pub fn values(&self) -> &PyList {
+        unsafe {
+            PySequence::try_from_unchecked(self.call_method0("values").unwrap())
+                .to_list()
+                .unwrap()
+        }
+    }
+
+    /// Returns a list of mappingproxy items.
+    ///
+    /// This is equivalent to the Python expression `list(mappingproxy.items())`.
+    pub fn items(&self) -> &PyList {
+        unsafe {
+            PySequence::try_from_unchecked(self.call_method0("items").unwrap())
+                .to_list()
+                .unwrap()
+        }
+    }
+
+    ///
+    /// Returns an iterator of `(key, value)` pairs in this mappingproxy.
+    pub fn iter(&self) -> PyMappingProxyIterator<'_> {
+        IntoIterator::into_iter(self)
+    }
+
+    /// Returns `self` cast as a `PyMapping`.
+    pub fn as_mapping(&self) -> &PyMapping {
+        unsafe { PyMapping::try_from_unchecked(self) }
+    }
+}
+
+pub struct PyMappingProxyIterator<'py> {
+    iterator: &'py PyIterator,
+    mappingproxy: &'py PyMappingProxy,
+}
+
+impl<'py> Iterator for PyMappingProxyIterator<'py> {
+    type Item = (&'py PyAny, &'py PyAny);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.iterator
+            .next()
+            .map(Result::unwrap)
+            .and_then(|key| self.mappingproxy.get_item(key).map(|value| (key, value)))
+    }
+}
+
+impl<'a> std::iter::IntoIterator for &'a PyMappingProxy {
+    type Item = (&'a PyAny, &'a PyAny);
+    type IntoIter = PyMappingProxyIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        PyMappingProxyIterator {
+            iterator: PyIterator::from_object(self).unwrap(),
+            mappingproxy: self,
+        }
+    }
+}
+
+/// Conversion trait that allows a sequence of tuples to be converted into `PyMappingProxy`
+/// Primary use case for this trait is `call` and `call_method` methods as keywords argument.
+pub trait IntoPyMappingProxy<'py> {
+    /// Converts self into a `PyMappingProxy` object pointer. Whether pointer owned or borrowed
+    /// depends on implementation.
+    fn into_py_mappingproxy(self, py: Python<'py>) -> PyResult<&'py PyMappingProxy>;
+}
+
+impl<'py, T, I> IntoPyMappingProxy<'py> for I
+where
+    T: PyDictItem,
+    I: IntoIterator<Item = T>,
+{
+    fn into_py_mappingproxy(self, py: Python<'py>) -> PyResult<&'py PyMappingProxy> {
+        let dict = self.into_py_dict(py);
+        PyMappingProxy::new(py, dict.as_mapping())
+    }
+}
```

Comment:
This is a cute trait, but I'm not really convinced this is necessary given the user can just do `PyMappingProxy::new(py, v.into_py_dict(py).as_mapping())`. Have you got a use case in mind?